### PR TITLE
docs: add a section on performance evaluation

### PR DESCRIPTION
**Pull request by mroavi**
Addresses: https://github.com/TensorBFS/TensorInference.jl/issues/68

### CODE REVIEW

**Comment by GiggleLiu** [comment]
Can we move the benchmark scripts and data files to another repo and only keep the svg files?

**Comment by mroavi** [comment]
> Can we move the benchmark scripts and data files to another repo and only keep the svg files?

Sure. I moved them to https://github.com/TensorBFS/TensorInferenceBenchmarks.jl/tree/main/scripts/performance-evaluation.

I actually went for the same approach of using Julia to generate the pgfplot graph so that we can tweak it easily if needed.

**Comment by mroavi** [comment]
I had issues with Julia generating the performance evaluation graph. I generated it offline and uploaded the svg to the docs' assets dir.

File: docs/src/performance-evaluation.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+```@raw html
+<img src="./assets/performance-evaluation.svg" width="700" style="margin-left: auto; margin-right: auto; display:block;">
+```
+
+The benchmark problems are arranged along the x-axis in ascending order of
+complexity, measured by the induced tree width. On average,
+*TensorInference.jl* achieves a speedup of 11 times across all problems.
+Notably, for the 10 most complex problems, the average speedup increases to 63
+times, highlighting its superior scalability.
```
Comment (GiggleLiu) [inline docs/src/performance-evaluation.md:25]: We need some description about the fitted line.

File: docs/make.jl [hunk 2]
```diff
@@ -58,6 +58,7 @@ makedocs(;
             "Asia Network" => "generated/asia-network/main.md",
             "Hard-core Lattice Gas" => "generated/hard-core-lattice-gas/main.md",
            ],
+        "Performance evaluation" => "performance-evaluation.md",
         "Performance tips" => "generated/performance.md",
```
Comment (GiggleLiu) [inline docs/make.jl:62]: Please also change the last line of performance tips. It mentioned the benchmark but with no link.

**Pull request merged**